src/core/featuremodel.cpp

```diff
@@ -469,6 +471,13 @@ void FeatureModel::applyGeometry()
     geometry.avoidIntersections( intersectionLayers );
   }
 
+  if ( mLayer && mLayer->geometryOptions()->geometryPrecision() != 0.0 )
+  {
+    const double precision = mLayer->geometryOptions()->geometryPrecision();
+    QgsGeometry snappedGeometry = geometry.snappedToGrid( precision, precision );
+    geometry = snappedGeometry.makeValid();
```

Comment:
Can you add a summary of the decisions for snapping precision to the PR description (so it can easily be ported to the docs)

Reply:
@m-kuhn , updated the PR description to add implementation details.